.github/workflows/build.yml

```diff
@@ -0,0 +1,96 @@
+name: Pull Sources from Manifest and build
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+env:
+  CI_ONIRO_MANIFEST_URL: ${{ github.server_url }}/${{ github.repository }}
+  CI_ONIRO_MANIFEST_NAME: default.xml
+  CI_ONIRO_MANIFEST_BRANCH: ${{ github.ref_name }}
+  CI_ONIRO_RUNNER_PERSISTENT_STORAGE: /home/openharmony/persistent_storage
+
+jobs:
+  pull_sources:
+    runs-on: self-hosted
+    container:
+      image: swr.cn-south-1.myhuaweicloud.com/openharmony-docker/openharmony-docker:1.0.0
+      volumes:
+        - /home/francesco/actions-runner/persistent_storage:/home/openharmony/persistent_storage
```

Comment:
Need to change when we using a more generic solution. Just a note, so we do not forget.

Reply:
updated for a more generic solution